x-pack/plugin/write-load-forecaster/src/test/java/org/elasticsearch/xpack/writeloadforecaster/LicensedWriteLoadForecasterTests.java

```diff
@@ -445,6 +455,72 @@ private void testShardChangeDoesNotChangeTotalForecastLoad(ShardCountChange shar
+        final WriteLoadForecaster writeLoadForecaster = new LicensedWriteLoadForecaster(hasValidLicense::get, threadPool, maxIndexAge);
+        writeLoadForecaster.refreshLicense();
+
+        final ProjectMetadata.Builder metadataBuilder = ProjectMetadata.builder(randomProjectIdOrDefault());
+        final String dataStreamName = "logs-es";
+        final int numberOfBackingIndices = 10;
+        final int numberOfShards = randomIntBetween(1, 5);
+        final int numberOfIndicesWithZeroUptime = randomBoolean()
+            ? numberOfBackingIndices
+            : randomIntBetween(1, numberOfBackingIndices - 1);
+        final Set<Integer> indicesWithZeroUptime = new HashSet<>(
+            randomSubsetOf(numberOfIndicesWithZeroUptime, IntStream.range(0, numberOfBackingIndices).boxed().collect(Collectors.toSet()))
+        );
+        logger.info(
+            "--> indices with zero uptime: {}/{} ({})",
+            numberOfIndicesWithZeroUptime,
+            numberOfBackingIndices,
+            indicesWithZeroUptime
+        );
+        final List<Index> backingIndices = new ArrayList<>();
+        for (int i = 0; i < numberOfBackingIndices; i++) {
+            final IndexMetadata indexMetadata = createIndexMetadata(
+                DataStream.getDefaultBackingIndexName(dataStreamName, i),
+                numberOfShards,
+                indicesWithZeroUptime.contains(i)
+                    ? indexWriteLoadWithMissingOrZeroUptime(numberOfShards)
+                    : randomIndexWriteLoad(numberOfShards),
+                System.currentTimeMillis() - (maxIndexAge.millis() / 2)
+            );
+            backingIndices.add(indexMetadata.getIndex());
+            metadataBuilder.put(indexMetadata, false);
+        }
+
+        final IndexMetadata writeIndexMetadata = createIndexMetadata(
+            DataStream.getDefaultBackingIndexName(dataStreamName, numberOfBackingIndices),
+            numberOfShards,
+            null,
+            System.currentTimeMillis()
+        );
+        backingIndices.add(writeIndexMetadata.getIndex());
+        metadataBuilder.put(writeIndexMetadata, false);
+
+        final DataStream dataStream = createDataStream(dataStreamName, backingIndices);
+        metadataBuilder.put(dataStream);
+
+        final ProjectMetadata.Builder updatedMetadataBuilder = writeLoadForecaster.withWriteLoadForecastForWriteIndex(
+            dataStream.getName(),
+            metadataBuilder
+        );
+        final IndexMetadata writeIndex = updatedMetadataBuilder.getSafe(dataStream.getWriteIndex());
+        final OptionalDouble forecastedWriteLoad = writeLoadForecaster.getForecastedWriteLoad(writeIndex);
+
+        final boolean someIndicesHadUptime = numberOfIndicesWithZeroUptime != numberOfBackingIndices;
+        assertThat(forecastedWriteLoad.isPresent(), is(someIndicesHadUptime));
+        if (someIndicesHadUptime) {
+            assertThat(forecastedWriteLoad.getAsDouble(), not(notANumber()));
+        }
+    }
```

Comment:
Could probably have tested this by calling `forecastIndexWriteLoad` directly (it's exposed for testing). Happy to do that instead if we want to reduce the boilerplate.